crates/telio-dns/src/forward.rs

```diff
@@ -32,21 +32,23 @@ use trust_dns_server::{
 use crate::bind_tun;
 
 #[derive(Clone, Copy)]
-pub struct TelioRuntime;
-impl RuntimeProvider for TelioRuntime {
+pub struct TelioRuntime<const USE_IPV6: bool>;
+impl<const USE_IPV6: bool> RuntimeProvider for TelioRuntime<USE_IPV6> {
```

Comment:
Do we really care if the forward authority uses IPv6 or not for resolving the DNS?

Reply:
There is this theoretical edge-case where forward server may potentially be IPv6. At some point we were working on (LLT-909) and one of the solutions was attempt to use another meshnet node's DNS server when routing traffic through it. This has not been released, and does not look like we are going to be releasing it any time soon. But at least theoretically in that case forward server could be IPv6. Now whether it makes sense to introduce IPv6 support in `forward.rs` for this edge-case, I do not know, but it seems like we already have some amount of it, so maybe removing is not really beneficial. What is more, looks like safe changes for me.

Reply:
Oh wait, but `forward.rs` had IPv6 support previously and it is already in prod. In that case I would agree with @dfetti that forward.rs does not need to be feature flagged here.